addons/stock_account/models/stock_move.py

```diff
@@ -57,13 +57,13 @@ def _get_in_move_lines(self):
         :rtype: recordset
         """
         self.ensure_one()
-        res = self.env['stock.move.line']
+        res = OrderedSet()
         for move_line in self.move_line_ids:
```

Comment:
could be a filter as whe suggested

Reply:
Nein! I am not sur that it will more readable:
```python
        self.ensure_one()
        def in_filter(move_line):
            if move_line.owner_id and move_line.owner_id != move_line.company_id.partner_id:
                return False
            elif not move_line.location_id._should_be_valued() and move_line.location_dest_id._should_be_valued():
                return True
            else:
                return False
        return self.move_line_ids.filtered(in_filter)
```

Reply:
![](https://media.giphy.com/media/sV7vFvd3BD1UScvRRA/giphy.gif)